native/python/include/jp_pythontypes.h

```diff
@@ -243,61 +203,39 @@ class JPPyString : public JPPyObject
  * Container types
  ***************************************************************************/
 
-/** Wrapper for a Python tuple object. */
-class JPPyTuple : public JPPyObject
-{
-public:
-
-	JPPyTuple(JPPyRef::Type usage, PyObject* obj) : JPPyObject(usage, obj)
-	{
-	}
-
-	JPPyTuple(const JPPyTuple &self) : JPPyObject(self)
-	{
-	}
-
-	JPPyTuple& operator = (const JPPyTuple &self)
-	{
-		JPPyObject::operator=(self);
-		return *this;
-	}
-
-	/** Create a new tuple holding a fixed number of items.
-	 *
-	 * Every item must be set before the tuple is used or we are heading
-	 * for a segfault.  Tuples are not mutable so items can only be set
-	 * during creation.
-	 */
-	static JPPyTuple newTuple(jlong sz);
-
-	/** Set an item in the tuple.
-	 *
-	 * This does not steal a reference to the object.
-	 */
-	void setItem(jlong ndx, PyObject* val);
-
-} ;
-
 /** Wrapper for a Python sequence.
  *
  * In most cases, we will not use this directly, but rather convert to
  * a JPPyObjectVector for easy access.
  */
-class JPPySequence : public JPPyObject
+class JPPySequence
 {
+	JPPyObject obj_;
+
+	JPPySequence(PyObject* obj)
+	{
+		obj_ = JPPyObject::use(obj);
+	}
+
 public:
 
-	JPPySequence(JPPyRef::Type usage, PyObject* obj) : JPPyObject(usage, obj)
+	JPPySequence(const JPPySequence& seq)
+	: obj_(seq.obj_)
 	{
 	}
 
-	JPPySequence(const JPPyObject &self) : JPPyObject(self)
+	static JPPySequence use(PyObject* obj)
 	{
+		return JPPySequence(obj);
 	}
 
 	JPPyObject operator[](jlong i);
 
 	jlong size();
+
+private:
+	JPPySequence& operator= (const JPPySequence& ) ;
```

Comment:
you allow copy construction, but forbid assignment *head scratching*

Reply:
I think I tried to remove both and one didn't work so I just put it back.  I will reexamine.

Reply:
This is a odd edge case of using named constructors.  In order to use a named constructor the class must have a copy constructor even through that copy constructor will not be called due to optimizations.   On the other hand assignment operator is not required.    

I am not sure why C++ has this formulation (and I have tested and found that the copy constructor is optimized out), but it is still required.